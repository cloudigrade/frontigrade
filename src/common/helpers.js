import _ from 'lodash';

const generateId = prefix => `${prefix || 'generatedid'}-${Math.ceil(1e5 * Math.random())}`;

const getErrorMessageFromResults = (results, filterField = null) => {
  const messageResponse = _.get(results, 'response.data', results.message);
  const detailResponse = _.get(results, 'response.data', results.detail);

  if (typeof messageResponse === 'string') {
    return messageResponse;
  }

  if (typeof detailResponse === 'string') {
    return detailResponse;
  }

  const getMessages = (messageObject, filterKey) => {
    const obj = filterKey ? messageObject[filterKey] : messageObject;

    return _.map(
      obj,
      next => {
        if (_.isArray(next)) {
          return getMessages(next);
        }

        return next;
      },
      null
    );
  };

  return _.join(getMessages(messageResponse || detailResponse, filterField), '\n');
};

const noop = Function.prototype;

const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state };

  if (prop && !state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (reset && prop && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset && prop) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else if (reset && !prop) {
    obj = {
      ...state,
      ...initialState,
      ...data
    };
  } else if (prop) {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  } else {
    obj = {
      ...state,
      ...data
    };
  }

  return obj;
};

const prettyPrintJson = json => JSON.stringify(json, null, 2);

const DEV_MODE = process.env.REACT_APP_ENV === 'development';

const OC_MODE = process.env.REACT_APP_ENV === 'oc';

const FULFILLED_ACTION = base => `${base}_FULFILLED`;

const PENDING_ACTION = base => `${base}_PENDING`;

const REJECTED_ACTION = base => `${base}_REJECTED`;

const helpers = {
  generateId,
  getErrorMessageFromResults,
  noop,
  setStateProp,
  prettyPrintJson,
  DEV_MODE,
  OC_MODE,
  FULFILLED_ACTION,
  PENDING_ACTION,
  REJECTED_ACTION
};

export { helpers as default, helpers };
