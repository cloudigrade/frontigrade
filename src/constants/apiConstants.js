const API_ACCOUNT_ARN = 'account_arn';
const API_ACCOUNT_NAME = 'name';
const API_RESOURCE_TYPE = 'resourcetype';
const API_AWS_ACCOUNT_ID = 'aws_account_id';

const apiTypes = {
  API_ACCOUNT_ARN,
  API_ACCOUNT_NAME,
  API_RESOURCE_TYPE,
  API_AWS_ACCOUNT_ID
};

export { apiTypes as default, apiTypes, API_ACCOUNT_ARN, API_ACCOUNT_NAME, API_RESOURCE_TYPE, API_AWS_ACCOUNT_ID };
