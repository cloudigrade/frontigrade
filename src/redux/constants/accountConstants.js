const ADD_ACCOUNT = 'ADD_ACCOUNT';
const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
const DELETE_ACCOUNTS = 'DELETES_ACCOUNTS';
const GET_ACCOUNT = 'GET_ACCOUNT';
const GET_ACCOUNTS = 'GET_ACCOUNTS';
const ADD_ACCOUNT_WIZARD = 'ADD_ACCOUNT_WIZARD';
const ADD_ACCOUNT_WIZARD_STEP_POLICY = 'ADD_ACCOUNT_WIZARD_STEP_POLICY';
const INVALID_ACCOUNT_WIZARD_STEP_POLICY = 'INVALID_ACCOUNT_WIZARD_STEP_POLICY';
const ADD_ACCOUNT_WIZARD_STEP_ROLE = 'ADD_ACCOUNT_WIZARD_STEP_ROLE';
const INVALID_ACCOUNT_WIZARD_STEP_ROLE = 'INVALID_ACCOUNT_WIZARD_STEP_ROLE';
const ADD_ACCOUNT_WIZARD_STEP_ARN = 'ADD_ACCOUNT_WIZARD_STEP_ARN';
const INVALID_ACCOUNT_WIZARD_STEP_ARN = 'INVALID_ACCOUNT_WIZARD_STEP_ARN';
const ADD_ACCOUNT_SHOW = 'ADD_ACCOUNT_SHOW';
const EDIT_ACCOUNT_SHOW = 'EDIT_ACCOUNT_SHOW';
const UPDATE_ACCOUNT_HIDE = 'UPDATE_ACCOUNT_HIDE';

export {
  ADD_ACCOUNT,
  DELETE_ACCOUNT,
  DELETE_ACCOUNTS,
  GET_ACCOUNT,
  GET_ACCOUNTS,
  ADD_ACCOUNT_WIZARD,
  ADD_ACCOUNT_WIZARD_STEP_POLICY,
  INVALID_ACCOUNT_WIZARD_STEP_POLICY,
  ADD_ACCOUNT_WIZARD_STEP_ROLE,
  INVALID_ACCOUNT_WIZARD_STEP_ROLE,
  ADD_ACCOUNT_WIZARD_STEP_ARN,
  INVALID_ACCOUNT_WIZARD_STEP_ARN,
  ADD_ACCOUNT_SHOW,
  EDIT_ACCOUNT_SHOW,
  UPDATE_ACCOUNT_HIDE
};
