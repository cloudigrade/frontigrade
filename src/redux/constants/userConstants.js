const USER_CREATE = 'USER_CREATE';
const USER_DELETE = 'USER_DELETE';
const USER_INFO = 'USER_INFO';
const USER_LOGIN = 'USER_LOGIN';
const USER_LOGOUT = 'USER_LOGOUT';

export { USER_CREATE, USER_DELETE, USER_INFO, USER_LOGIN, USER_LOGOUT };
