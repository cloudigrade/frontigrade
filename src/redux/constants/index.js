import * as toastNotificationTypes from './toastNotificationConstants';
import * as userTypes from './userConstants';

const reduxTypes = {
  toastNotifications: toastNotificationTypes,
  user: userTypes
};

export { reduxTypes as default, reduxTypes, toastNotificationTypes, userTypes };
