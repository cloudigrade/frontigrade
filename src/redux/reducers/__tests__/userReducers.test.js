import { userReducers } from '../';
import { userTypes as types } from '../../constants';
import helpers from '../../../common/helpers';

describe('UserReducers', () => {
  it('should return the initial state', () => {
    expect(userReducers.initialState).toBeDefined();
  });

  it('should handle all defined error types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: helpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = userReducers(undefined, dispatched);

      expect({ type: helpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot();
    });
  });

  it('should handle all defined pending types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: helpers.PENDING_ACTION(value)
      };

      const resultState = userReducers(undefined, dispatched);

      expect({ type: helpers.PENDING_ACTION(value), result: resultState }).toMatchSnapshot();
    });
  });

  it('should handle all defined fulfilled types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          response: {
            data: {
              test: 'success'
            }
          }
        }
      };

      const resultState = userReducers(undefined, dispatched);

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot();
    });
  });
});
