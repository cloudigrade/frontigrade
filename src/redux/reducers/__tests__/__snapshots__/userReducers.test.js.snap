// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReducers should handle all defined error types 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_CREATE_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types 2`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_DELETE_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types 3`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_INFO_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types 4`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "loginFailed": true,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGIN_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types 5`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGOUT_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types 6`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REJECTED",
}
`;

exports[`UserReducers should handle all defined error types 7`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": true,
      "errorMessage": "MESSAGE",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REMOVE_REJECTED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "userInfo": undefined,
    },
  },
  "type": "USER_CREATE_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 2`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "userInfo": undefined,
    },
  },
  "type": "USER_DELETE_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 3`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": undefined,
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_INFO_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 4`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGIN_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 5`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGOUT_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 6`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loginFailed": false,
      "pending": false,
      "remember": true,
      "storedEmail": undefined,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_FULFILLED",
}
`;

exports[`UserReducers should handle all defined fulfilled types 7`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REMOVE_FULFILLED",
}
`;

exports[`UserReducers should handle all defined pending types 1`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "userInfo": Object {},
    },
  },
  "type": "USER_CREATE_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types 2`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": false,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "userInfo": Object {},
    },
  },
  "type": "USER_DELETE_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types 3`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": true,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_INFO_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types 4`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": true,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGIN_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types 5`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": true,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_LOGOUT_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types 6`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": true,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_PENDING",
}
`;

exports[`UserReducers should handle all defined pending types 7`] = `
Object {
  "result": Object {
    "session": Object {
      "authorized": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "loginFailed": false,
      "pending": true,
      "remember": false,
      "storedEmail": null,
      "userInfo": Object {},
    },
    "user": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "userInfo": Object {},
    },
  },
  "type": "USER_STORED_DATA_REMOVE_PENDING",
}
`;
